### DISCUSSION TITLE
Aula 7 - Listas Assíncronas  -  Criando Apps Web com Angular e UiKit

### DISCUSSION THREAD

**Discussion by leosilka** [balta-io/forum#425] [Tira Dúvidas] [answered]
Boa tarde, estou enfrentando um probleminha que me travou agora, ao usar o `public products$: Observable<any[]>` gera erro no vscode: "A propriedade 'products$' não tem nenhum inicializador e não está definitivamente atribuída no construtor."

Pois bem, tentei procurar alguma solução para isso, usei ! na frente do **products$** mas sem sucesso, segue meu código abaixo:
![image](https://github.com/balta-io/forum/assets/101304835/c765c4c6-034a-4d02-9729-5f7efe447f97)

E mesmo se eu usar ! na frente, o erro ele desaparece, porém, a lista de itens não carrega, segue abaixo o código na page:
![image](https://github.com/balta-io/forum/assets/101304835/ded7c750-90d7-4cf2-a726-b083adccc527)

Se eu usar o *ngFor ele desaparece todo o restante, um exemplo é o botão adicionar.
![image](https://github.com/balta-io/forum/assets/101304835/9ac7cae4-b027-44e6-85e6-b6df6ec15946)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @leosilka, como vai?
Você está usando as mesmas versões do curso?

**Comment by leosilka** [comment]
Bom dia Professor @andrebaltieri , estou usando o Angular 16 neste curso.
Será que se eu instalar a mesma versão do curso, terei que refatorar o código por completo?

**Reply by andrebaltieri** [reply]
Bom dia, @leosilka, como vai?
Recomendo que utilize as mesmas versões do curso!

**Reply by leosilka** [reply]
Belezinha, vou refazer as instalações das versões do curso professor.
